web/src/components/repo/settings/SecretsTab.vue

```diff
@@ -69,7 +69,25 @@ async function loadSecrets(page: number): Promise<Secret[] | null> {
     throw new Error("Unexpected: Can't load repo");
   }
 
-  return apiClient.getSecretList(repo.value.id, page);
+  const secrets: Record<string, Secret & { edit?: boolean }> = {};
+
+  // TODO: properly handle pagination
```

Comment:
That would be possible. We only need to think how we handle overwrites then.

Reply:
> Either show them with a separate badge or just skip them and don't show.

I already overwrite them with lower levels, this however only works if the top-level is in the list and the lower level is loaded as well. Otherwise it would for example show that a global secret `my-secret` exists and would only later when the repo-secrets are loaded show the overloading `my-secret` from the repo.

Reply:
> How about: Reaching the end of repo pages loads org pages and on this end load global ones. (This would need an additional callback of the pagination helper to notifiy that the last page is reached)

That sounds a bit tricky. Just thinking about it, isn't the current approach already working fine? It starts getting page 1 from repo, org and global, continues with page 2 from all, as soon as one of them is empty it will add 0 items to the list for all ongoing pages. It ends as soon as all 3 add 0 items.

Reply:
I actually think it shouldn't as it would show for example 3*10 secrets and in case in those two have the same name it would overwrite them. If two secrets with the same name and different levels appear on different pages it would probably show both.

Reply:
Imho it should be resolved at API side to not show global/org secrets that are overwritten. Also on the UI side would be nice to have the option to hide secrets that are inherited

Reply:
That's how I did it first, but actually reverted it as it made the api so ugly and involved quite some code to check for permissions.

Reply:
I adjusted the mapping again. As secrets are sorted by their name it should work now.